schema/deploy/functions/hamming_distance.sql

```diff
@@ -28,6 +28,34 @@ create or replace function public.hamming_distance(a text, b text)
     immutable
     parallel safe;
 
+create or replace function public.hamming_distance_lte(a text, b text, threshold integer)
```

Comment:
Because this will not return the actual hamming distance between the two strings if the actual is more than the threshold, it could be better to return a boolean .... it's a flag ... "yes or no, was the hamming distance between the two values greater than the threshold?. If someone wanted the actual hamming distance they could use the original function.

select hamming_distance_lte('AAAAAAAA', 'BBBBBBBB', 3)  returns 4 when the actual hamming distance is 8. 
We really just need to know that yes, this was above threshold.

Reply:
For our distance check, yes, we only need a boolean. For general querying, though, there are times when I've wanted to see the distance of close-enough barcodes. Returning the distance helps in that case because you can compute the distance just once instead of twice.

I agree that the boolean comparison feels slightly clearer than the threshold comparison:

    hamming_distance_lte(a, b, d) <= d
    hamming_distance_lte(a, b, d) is true

…but not clearer enough for me to feel it offsets the double-run downside. I doubt there is a speed difference between the two comparisons.

There's precedent for returning the thresholded distance with the similar [`levenshtein_less_equal()` function](https://www.postgresql.org/docs/10/fuzzystrmatch.html#id-1.11.7.25.6).

Reply:
OK. I was coming from the simplicity and clearness perspective.